### ISSUE TITLE
[css-images] Define what happens when all image-set options are invalid

### ISSUE THREAD

**Issue by emilio** [w3c/csswg-drafts#8266]
https://drafts.csswg.org/css-images-4/#image-set-notation says:

> First, remove any <image-set-option>s from the list that specify an unknown or unsupported MIME type in their type() value.
> [...]
> Finally, among the remaining [<image-set-option>](https://drafts.csswg.org/css-images-4/#typedef-image-set-option)s, make a UA-specific choice of which to load, based on whatever criteria deemed relevant (such as the resolution of the display, connection speed, etc).

What if there are no remaining options? Firefox chooses just the first [here](https://searchfox.org/mozilla-central/rev/8fd39dc89c68501a354446a8681ec8263d15e5aa/servo/components/style/values/computed/image.rs#84) but that's fairly arbitrary as well.

**Comment by SebastianZ** [comment]
I guess, the options in that case are:

1. Let the UA choose an arbitrary one (like Firefox does).
2. Specify which one to choose.
3. Don't choose any option. I.e. no image is displayed.

Sebastian

**Comment by rreno** [comment]
I think it makes the most sense to not display any image if there are no remaining options. If fallback behavior is wanted the author can use the `image` notation like this:

```CSS
.some-class {
    background-image: image(image-set(...), image("fallback.png"));
}
```

**Comment by rreno** [comment]
I think we should choose not to load any image if there are no options remaining after filtering by type.

>If the [[\<string\>]](https://www.w3.org/TR/css-values-4/#string-value), when parsed as a [valid MIME type string](https://mimesniff.spec.whatwg.org/#valid-mime-type), is either not valid, or is valid but doesn’t specify a supported image format, the [[\<image-set-option\>]](https://w3c.github.io/csswg-drafts/css-images-4/#typedef-image-set-option) does not define a valid option. (This has no effect on the validity of the [image-set()](https://w3c.github.io/csswg-drafts/css-images-4/#funcdef-image-set) function.)

Since the MIME type string doesn't determine whether or not the `image-set` function itself is valid we shouldn't fall back to a previous value for the property.

**Comment by rreno** [comment]
I made a PR to change the tests in WPT https://github.com/web-platform-tests/wpt/pull/38662

Please have a look! I can't add reviewers manually.

**Comment by emilio** [comment]
I'd prefer to have a resolution or a quick ack from the spec editors before changing the test again, but sure, looks good if we decide to go that route, and it seems sensible.

I don't think it matters much in practice between choosing first or choosing not to render an image because the only way to get an empty set but still render something is to lie in the type() function.

Still explicitly choosing nothing seems less confusing, so I'd support that.

@tabatkins @fantasai @LeaVerou can you confirm? Thanks!

**Comment by tabatkins** [comment]
Yeah if we discard all the options I'd expect it to be an "invalid image".
